static/js/cf-frontend.js

```diff
@@ -264,25 +267,30 @@ var appMode = 'edit'; // 'edit' or 'display'. also support
 // sets globals such as rawInputLst, code input box, and toggle options
 function parseQueryString() {
   var queryStrOptions = getQueryStringOptions();
-  if (queryStrOptions.preseededCode) {
-    pyInputSetValue(queryStrOptions.preseededCode);
+  //parse url & set corresponding values
+  if (queryStrOptions.precededCode) {
+    pyInputSetValue(queryStrOptions.precededCode);
+  }
+  if(queryStrOptions.type){
+    $("#type_system").val(queryStrOptions.type);
   }
   // ugh tricky -- always start in edit mode by default, and then
   // switch to display mode only after the code successfully executes
   appMode = 'edit';
   if ((queryStrOptions.appMode == 'display' ||
        queryStrOptions.appMode == 'visualize' /* 'visualize' is deprecated */) &&
-      queryStrOptions.preseededCode /* jump to display only with pre-seeded code */) {
-    executeCode(queryStrOptions.preseededCurInstr); // will switch to 'display' mode
+      queryStrOptions.precededCode /* jump to display only with pre-seeded code */) {
+    executeCode(queryStrOptions.precededCodeCurInstr); // will switch to 'display' mode
   }
   $.bbq.removeState(); // clean up the URL no matter what
 }
 
 // parsing the URL query string hash
 function getQueryStringOptions() {
-  return {preseededCode: $.bbq.getState('code'),
-          preseededCurInstr: Number($.bbq.getState('curInstr')),
-          appMode: $.bbq.getState('mode'),
+  return {precededCode: $.bbq.getState('code'),
+          type: $.bbq.getState('type'),
```

Comment:
`type` is not clear. Maybe `typeSystem` is better to imply this parameter is for indicating type system.